CMakeLists.txt

```diff
@@ -18,6 +18,10 @@ else ()
 endif ()
 
 add_library(libtelnet libtelnet.c)
+set_target_properties(libtelnet
+  PROPERTIES
+  VERSION ${PROJECT_VERSION}
+  SOVERSION ${PROJECT_VERSION})
```

Comment:
**NAK** on this line: current master branch uses `-version-info 2:0:0`, so using PROJECT_VERSION would result in 0.30.0 which cannot be desired...

Reply:
We can change it to `2` if you prefer, it's just that the current situation is just as bad, namely that it adds no `SOVERSION`

Reply:
The proper way would be to check whether/how the ABI changed... I agree that the current situation is bad and I would call it a regression compared to the former autotools approach... but when adding `SOVERSION` now we should take the time required to make it a real solution...

Reply:
Repushed, now:

    readelf -d temp/usr/local/lib64/liblibtelnet.so | grep SONAME
    0x000000000000000e (SONAME)             Library soname: [liblibtelnet.so.2]

Which mirrors the output of `-version-info 2:0:0`

Reply:
I could live with this, thanks :+1: